app/src/main/java/com/onegini/mobile/exampleapp/adapter/PendingPushMessagesAdapter.java

```diff
@@ -77,27 +80,44 @@ public void onBindViewHolder(final PendingPushMessagesAdapter.ViewHolder viewHol
 
     calendar.add(Calendar.SECOND, oneginiMobileAuthWithPushRequest.getTimeToLiveSeconds());
     viewHolder.expiresTextView.setText(context.getString(R.string.notification_expires_at, sdf.format(calendar.getTime())));
+
+    viewHolder.onClickListener = v -> context.startService(getServiceIntent(oneginiMobileAuthWithPushRequest));
   }
 
   @Override
   public int getItemCount() {
     return list.size();
   }
 
+  private Intent getServiceIntent(final OneginiMobileAuthWithPushRequest mobileAuthWithPushRequest) {
```

Comment:
this block of code is duplicated in the `FCMListenerService.java`, maybe you should extract it somewhere and also a there a method for parsing the intent to a `OneginiMobileAuthWithPushRequest`? SOme kind of transactionParser class or something similar...